### ISSUE TITLE
Request Feature

### ISSUE THREAD

**Issue by Esmaeili-Najafabadi** [GavriYashar/Matlab-Editor-Plugin#86]
Like intellij it is better to have ctrl+shift+Enter option which completes the line and go to the next line.

**Comment by GavriYashar** [comment]
if i under stand correctly it should work like this:

type:
``` Matlab
function myFunc(in
```
then press <kbd>CTRL</kbd> + <kbd>SHIFT</kbd> + <kbd>ENTER</kbd>
and then it will insert closing parenthesis and `end`
```
``` Matlab
function myFunc(in)
    % caret
end
```

**Comment by Esmaeili-Najafabadi** [comment]
Yes, exactly. Also automatically complete insert parentheses and semicolon at the end of a line.

**Comment by Esmaeili-Najafabadi** [comment]
One more question, why not just trying to customize intellij for matlab and running from there. It should be much simpler than what are you doing. Isn't it?

**Comment by GavriYashar** [comment]
> why not just trying to customize intellij for matlab and connecting the two platforms
i thought of it for a second, but there are a few but HUGE problems

- tab completion
  - especially custom objects or functions, so indexing is a thing now
  - then in matlab an expression like `true = false` or `true = someMatlabObject` so a correct 
tabcompletion is... hard
  - it is not guaranteed to know the type of a variable since it can be anything and overridden anytime with again.. anything
- custom language plugin for intellij
  - since some weird overloading is possible the colors would be buggy

> It should be much simpler than what are you doing. Isn't it?

it's way easier and __less time consuming__ (for me at least) to go the other way round (java to matlab, instead matlab to java)

**Comment by Esmaeili-Najafabadi** [comment]
You surely now that there are an autocomplete on the Intellij. Can't you just use that function? Besides, there are plenty of other functionalities that make intellij unique. 
I thought that working on a plugin for intellij would work better.
I also very searched for one but couldn't find something decent.:-(

**Comment by GavriYashar** [comment]
auto-complete relies on proper indexing. I'm not able to index `m` files correctly and use it proper on a variable I don't know the type of.

**Comment by Esmaeili-Najafabadi** [comment]
It is strange. since the pycharm surely do the same thing, without knowing the type of variable.
nevertheless, you may work on a plugin for pycharm instead of intellij.

**Comment by GavriYashar** [comment]
maybe i'm missing something. it may be easier than i think.

**Comment by Esmaeili-Najafabadi** [comment]
there are some inactive github repos on this. you may take a look at them. see also the ones for octave. 
Also some plugings for other languages. If you can find a decent plugin for another language on github you may get an idea how much work it needs. 
Anyway there are really two things of future that I miss today:
1) A fully functional Intellij latex plugin.
2) A fully functional Intellij Matlab plugin. :-)
